.circleci/config.yml

```diff
@@ -312,6 +313,75 @@ jobs:
+          command: |
+            wget https://github.com/juj/emsdk/archive/master.tar.gz
+            tar -xf master.tar.gz
+            cd emsdk-master
+            ./emsdk --notty update-tags
+            export WATERFALL_LKGR=`python -c "print [x.split('\"')[3] for x in open('upstream/lkgr.json').readlines() if '\"build\"' in x][0]"`
+            echo "LKGR: $WATERFALL_LKGR"
+            ./emsdk --notty install node-8.9.1-64bit # TODO: should emsdk install this automatically?
+            ./emsdk --notty install clang-upstream-$WATERFALL_LKGR-64bit
+            ./emsdk --notty activate clang-upstream-$WATERFALL_LKGR-64bit
+            cd -
+            # Use LLVM, clang, lld binaries from the emsdk
+            export LLVM="$HOME/emsdk-master/upstream/$WATERFALL_LKGR/bin"
+            # Remove the emsdk version of emscripten to save space in the
+            # persistent workspace and to avoid any confusion with that version
+            # we are trying to test.
+            rm -r ~/emsdk-master/upstream/$WATERFALL_LKGR/emscripten/
+            # Use Binaryen from the emsdk's waterfall (wasm backend can't use ports)
+            echo BINARYEN_ROOT="'$HOME/emsdk-master/upstream/$WATERFALL_LKGR/'" >> ~/.emscripten
+            # FIXME fix node
+            echo NODE_JS="'$HOME/emsdk-master/node/8.9.1_64bit/bin/node'" >> ~/.emscripten
+            echo "COMPILER_ENGINE = NODE_JS" >> ~/.emscripten
+            echo "JS_ENGINES = [NODE_JS]" >> ~/.emscripten
+            echo "==="
+            cat ~/.emscripten
+            echo "==="
+            python3 ~/emscripten/emcc.py -v
+      - run:
+          name: embuilder build ALL
+          command: |
+            # FIXME not all stuff builds in wb:
+            #   - al: see https://bugs.llvm.org/show_bug.cgi?id=39591
+            #   - cocos2d: source code error, cannot detect platform
+            EMCC_CORES=4 python3 ~/emscripten/embuilder.py build libc libc-mt libc-extras struct_info emmalloc dlmalloc dlmalloc_threadsafe pthreads libcxx libcxx_noexcept libcxxabi gl bullet freetype libpng ogg sdl2 sdl2-image sdl2-ttf sdl2-net vorbis zlib wasm-libc wasm_compiler_rt emmalloc_debug dlmalloc_debug dlmalloc_threadsafe_debug
+            python3 ~/emscripten/tests/runner.py test_hello_world
+      - persist_to_workspace:
+          # Must be an absolute path, or relative path from working_directory
+          root: ~/
+          # Must be relative path from root
+          paths:
+            - emsdk-master/
+            - .emscripten_cache/
+            - .emscripten_ports/
+            - .emscripten
+  test-waterfall-binaryen2:
+    <<: *test-defaults
+    environment:
+      - TEST_TARGET=binaryen2
+  test-waterfall-wasmobj2:
+    <<: *test-defaults
+    environment:
+      - TEST_TARGET=wasmobj2
+  test-waterfall-other:
+    <<: *test-defaults
+    environment:
+      - TEST_TARGET=other skip:other.test_native_link_error_message skip:other.test_emcc_v skip:other.test_llvm_lit
```

Comment:
Should we make these skips be `@no_wasm_backend` annotations in test_other.py?

Reply:
Its not really the wasm backend that makes the lit test fail.. its more like it can't find the llvm sources, which is true also for an installed fastcmp. 

Seems like running lit tests from here is a little strange.. should really be done when building fastcmp itself, no?  Should we remove test_llvm_lit completely?

Reply:
It's a little odd, yeah, but the reason is that we don't build fastcomp on CI (we use a prebuilt binary). This gives us coverage despite that.

I'd say adding an annotation in the python files makes sense for that one. For the other two, the issues are CI related, similar to the fastcomp testing, I'll add a comment.